types/index.d.ts

```diff
@@ -58,10 +58,27 @@ interface PlopActionHooks {
 	onFailure?: (failure: PlopActionHooksFailures) => void;
 }
 
+export type PromptAnswers = Record<string, any>;
+export type DynamicPromptsFunction = (inquirer: inquirer.Inquirer) => Promise<PromptAnswers>;
+export type DynamicActionsFunction = (data?: PromptAnswers) => ActionType[];
+
+export type Prompts = DynamicPromptsFunction | inquirer.Question[]
+	| inquirer.CheckboxQuestion[]
+	| inquirer.ListQuestion[]
+	| inquirer.ExpandQuestion[]
+	| inquirer.ConfirmQuestion[]
+	| inquirer.EditorQuestion[]
+	| inquirer.RawListQuestion[]
+	| inquirer.PasswordQuestion[]
+	| inquirer.NumberQuestion[]
+	| inquirer.InputQuestion[];
```

Comment:
one thought that comes to my mind here is that this limits the prompts array to always be of a specific type of question.

I'm going to change this to something like this so you can have different types of questions within the same array 

```ts
type PromptQuestion = inquirer.Question
  | /** other types */

type Prompts = DynamicPromptsFunction | PromptQuestion[];
```

Reply:
updated this :)